Makefile

```diff
@@ -138,60 +131,13 @@ distclean:
 	git clean -xdff
 	git submodule foreach git clean -xdff
 
-# $(call RUN_NUNIT_TEST,filename,log-lref?)
-define RUN_NUNIT_TEST
-	MONO_TRACE_LISTENER=Console.Out \
-	USE_MSBUILD=$(if $(USE_MSBUILD),$(USE_MSBUILD),0) \
-	$(RUNTIME) --runtime=v4.0.0 \
-		$(NUNIT_CONSOLE) $(NUNIT_EXTRA) $(1) \
-		$(if $(TEST),--test=$(TEST)) \
-		--result="TestResult-$(basename $(notdir $(1))).xml;format=nunit2" \
-		-output=bin/Test$(CONFIGURATION)/TestOutput-$(basename $(notdir $(1))).txt \
-	|| true ; \
-	if [ -f "bin/Test$(CONFIGURATION)/TestOutput-$(basename $(notdir $(1))).txt" ] ; then \
-		cat bin/Test$(CONFIGURATION)/TestOutput-$(basename $(notdir $(1))).txt ; \
-	fi
-	$(MAKE) rename-test-cases RENAME_GLOB="`pwd`/TestResult-$(basename $(notdir $(1))).xml"
-endef
-
-run-nunit-tests: $(NUNIT_TESTS)
+run-nunit-tests:
 ifeq ($(SKIP_NUNIT_TESTS),)
-	$(foreach t,$(NUNIT_TESTS), $(call RUN_NUNIT_TEST,$(t),1))
+	$(MSBUILD) $(MSBUILD_FLAGS) $(TEST_TARGETS) /t:RunNUnitTests
```

Comment:
One problem here is that it changes the semantics. This used to work:

    make run-nunit-tests TEST=insert-test-name-here

That will no longer work, because the `RunNUnitTests` target looks for an *environment variable*, and -- afaik -- `make` command-line variables aren't exported as environment variables in the child process.

As such, *I believe* that this line needs to be changed to:

    $(MSBUILD) $(MSBUILD_FLAGS) $(TEST_TARGETS) /t:RunNUnitTests $(if $(TEST),/p:TEST=$(TEST))